---
title: "Pre-implantation embryo development in human and mouse"
author: "Yutong Wang"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning = FALSE,
  message = FALSE,
  comment = "#>"
)
```


We illustrate the CorGI feature selection method by analyzing datasets on human and mouse pre-implantation embryo development.

```{r}
library(corgi)
library(ggplot2)
library(cowplot)
library(scmap)
library(scran)
library(M3Drop)
```

# Preprocessing the data

We download the data from the [Hemberg Lab website](https://hemberg-lab.github.io/scRNA.seq.datasets/). The human dataset is from [Yan et al 2013](http://dx.doi.org/10.1038/nsmb.2660) while the mouse dataset is from [Deng et al 2014](http://dx.doi.org/10.1126/science.1245316).

```{r, eval = F}
yan <- readRDS(url("https://scrnaseq-public-datasets.s3.amazonaws.com/scater-objects/yan.rds"))
deng <- readRDS(url("https://scrnaseq-public-datasets.s3.amazonaws.com/scater-objects/deng-reads.rds"))
```

```{r, include = F}
yan <- readRDS(file = "~/CorGI_RDS/embryo_devel/yan.rds")
deng <- readRDS(file = "~/CorGI_RDS/embryo_devel/deng-reads.rds")
```


Get genes common to both datasets
```{r}
rownames(yan) <- toupper(rownames(yan))
rownames(deng) <- toupper(rownames(deng))
shared_genes <- intersect(rownames(yan), rownames(deng))
```


Get rid of the spike-in genes
```{r}
shared_genes <- shared_genes[-grep("ERCC", shared_genes)]
```

Subset the data
```{r}
yan <- yan[shared_genes, ]
deng <- deng[shared_genes, ]
```

# Run CorGI feature selection

Normalization is not necessary for running CorGI

```{r, eval = F}
corgi_output_embyro_devel <-
  run_corgi(logcounts(yan), logcounts(deng))
```

```{r,eval=F,include=F}
devtools::use_data(corgi_output_embyro_devel, overwrite = T) # Cache the output so that we don't have to actually rerun corgi. Change to "eval=T" if rerun is necessary.
```

```{r,include=F}
data("corgi_output_embyro_devel") # Load the cached output
head(corgi_output_embyro_devel)
```

Select top CorGI genes
```{r}
corgi_scores <- corgi_output_embyro_devel
corgi_scores <- corgi_scores[,1:3]/corgi_scores[,4]
```

```{r}
plot(corgi_scores[ , "principal_curve"], corgi_scores[ , "kmmd_statistic"])
hist(corgi_scores[ , "kmmd_statistic"], breaks = 100)
median(corgi_scores[ , "kmmd_statistic"]) + mad(corgi_scores[ , "kmmd_statistic"])
```

```{r}
good <- corgi_scores[,"kmmd_statistic"] < Inf
# good <- corgi_scores[,"kmmd_statistic"] < median(corgi_scores[,"kmmd_statistic"]) +mad(corgi_scores[,"kmmd_statistic"])
# median(corgi_scores[,"kmmd_statistic"]) +mad(corgi_scores[,"kmmd_statistic"])
# good <- corgi_scores[,"kmmd_statistic"] < 0.33
corgi_gene_set <- top_n_genes(-corgi_scores[good ,"principal_curve"], 100)
```

# Visualization
Normalization

Metadata for the combined datasets
```{r}
goolam <- readRDS(file = "~/Downloads/goolam.rds")

rownames(goolam) <- toupper(rowData(goolam)$feature_symbol)

combined <- 
  combine_sces(
    sce_list = list(Yan = yan, Deng = deng, Goolam = goolam),
    levels = c("zygote", "2cell", "4cell", "8cell", "16cell", "blast")
  )

```

Wrapper around the qplot `ggplot2::qplot` function with our color scheme

```{r}
my_color_palette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
my_shape_palette <- c(16,1, 2)
qplot <- function(...){
  ggplot2::qplot(...) +
  scale_color_manual(values = my_color_palette) +
  scale_shape_manual(values = my_shape_palette)
}
```

```{r}
mds <- spearman_rho_mds(counts(combined)[intersect(corgi_gene_set,rownames(combined)), ])

qplot(mds[,1],mds[,2], shape = combined$batch, color = combined$cell_type)


```

## Principal component analysis (PCA)

To use PCA, we first need to adjust the read-depth to be comparable across the two datasets
```{r}
library(scran)
out <- multiBatchNorm(yan, deng, assay.type = "logcounts")
yan <- out[[1]]
deng <- out[[2]]
```



```{r}
HDG_ranking <- function(sce) {
  rowData(sce)$feature_symbol <- rownames(sce)
  sce <- scmap::selectFeatures(sce)
  return(rownames(sce)[order(rowData(sce)[["scmap_scores"]], decreasing = T, na.last = T)])
}

n_genes <- length(corgi_gene_set)

gene_sets <- get_compared_gene_sets(
  batch1_top_genes = HDG_ranking(yan),
  batch1_name = "HDG(Yan)",
  batch2_top_genes = HDG_ranking(deng),
  batch2_name = "HDG(Deng)",
  desired_size = n_genes
)

gene_sets[["CORGI"]] <- corgi_gene_set

names(gene_sets)
```

Reorder the gene sets
```{r}
gene_sets <- gene_sets[c(5, 1, 2, 3, 4)]
lapply(gene_sets, length)
```

```{r}
library(Seurat)
CCA_gene_loading <- Seurat_CCA_GeneLoading(
  normcounts(sample_cells(yan,50)),
  counts(sample_cells(deng,50)),
  k = 3)

leverage_scores <- rowSums(CCA_gene_loading^2)
gene_sets[["CCA-GL"]] <- top_n_genes(leverage_scores, n_genes)
```



```{r}
names(gene_sets)
```


```{r, include = F}
saveRDS(gene_sets, file = "~/CorGI_RDS/embryo_devel/gene_sets.rds")
```

```{r}
get_embeddings <- function(gene_sets){
  lapply(
      X = gene_sets,
      FUN = function(gene_set) {
        genes_use <- intersect(gene_set, rownames(combined))
        spearman_rho_mds(counts(combined)[genes_use, ])
      }
    )
}
```



```{r}
emb_name <- "MDS"

get_top_panel <- function(embeddings){
  get_scatterplots(embeddings,
                   as.factor(combined$batch),
                   combined$cell_type) -> dr_plots
  
  plot_grid(
    get_axes_legend(emb_name),
    get_shape_legend(combined$batch, my_shape_palette),
    get_color_legend(combined$cell_type, my_color_palette, ncol = 1),
    ncol = 1,
    rel_heights = c(1,1,2)
  ) -> legends
  
  # dr_plots[[length(dr_plots) + 1]] <- legends
  
  dr_plots[["nrow"]] <- 2
  
  dr_plots[["labels"]] <- c(LETTERS[1:6], "")
  
  do.call(plot_grid, dr_plots) -> top_panel
  plot_grid(top_panel, legends, nrow=1, rel_widths = c(4,1))
}

get_top_panel(get_embeddings(gene_sets[
  c("CORGI",
    "HDG(Deng)",
    "HDG(Yan)",
    "CCA-GL",
    "Union",
    "Intersection")])) -> embryo_devel_MDS_HDG

ggsave(filename = "embryo_devel_MDS_HDG.eps",
       device = "eps", 
       plot = embryo_devel_MDS_HDG,path = "~/CorGI_figures/embryo_devel", 
       width = 8,
       height = 4.5)




```

# Blakeley gene sets


- Pou5f1/POU5F1 (human cluster 49 and __mouse cluster 4__) shows an upregulation of expression from the 4-cell to the blastocyst stage.


- Mouse Sox2 expression was highly upregulated at the blastocyst stage (__cluster 36__) and, interestingly, was co-expressed with a number of TE-associated genes, including Gata2, Id2, Elf5 and Eomes.

- Whereas human NANOG expression was upregulated at the 4- to 8-cell stage (cluster 7), mouse Nanog expression was upregulated earlier between the zygotic and 2-cell stage (__cluster 1__). 

```{r}
library(openxlsx)
temp <- tempfile()
download.file("http://www.biologists.com/DEV_Movies/DEV123547/TableS1.xlsx",temp)
read.xlsx(temp) -> df
unlink(temp)

lapply(unique(df$Human.cluster.number),
       function(i){
         toupper(df$X1[df$Human.cluster.number==i])
       }) -> human_gene_sets
names(human_gene_sets) <- paste0("HC",unique(df$Human.cluster.number))

temp <- tempfile()
download.file("http://www.biologists.com/DEV_Movies/DEV123547/TableS2.xlsx",temp)
read.xlsx(temp) -> df
# read.xlsx("TableS2.xlsx") -> df
unlink(temp)

lapply(unique(df$Mouse.cluster.number),
       function(i){
         toupper(df$X1[df$Mouse.cluster.number==i])
       }) -> mouse_gene_sets
names(mouse_gene_sets) <- paste0("MC",unique(df$Mouse.cluster.number))

blakeley_gene_sets <- c(human_gene_sets[c(49, 14, 7 )], mouse_gene_sets[c(4, 36, 1)])
names(blakeley_gene_sets) <- c("POU5F1", "SOX2", "NANOG", "Pou5f1", "Sox2", "Nanog")

```

```{r}
# stop()
gene_set_use <- "CorGI"
gene_set <- intersect(gene_sets[[gene_set_use]], rownames(combined))

blakeley <- get_membership_labels_from_set_list(gene_set, blakeley_gene_sets)


emb <- get_embeddings(list(gene_set=gene_set))[[1]]
pcurve_out <- princurve::principal_curve(emb)



R <- 
  apply(combined[gene_set,],
        2,
        rank)
is_in_leverage <- rownames(R) %in% gene_sets[["Leverage"]]


```



```{r}

col_ann <- data.frame(cell_type = cell_type,
                      batch = batch,
                      pseudo_time = pcurve_out$lambda)

row_ann <- data.frame(blakeley = blakeley, 
                      is_in_leverage = is_in_leverage,
                      leverage_score = leverage_scores[rownames(R)])

cell_type_colors <- my_color_palette[1:6]
names(cell_type_colors) <- levels(cell_type)

annotation_colors <- list(cell_type = cell_type_colors,
                          leverage_score = c("white", "black"),
                          pseudo_time = c("white","black"))
rownames(col_ann) <- colnames(R)
```


```{r, fig.height=3}
plot_gene_by_cell_heatmap(
  R,
  annotation_col = col_ann,
  order_cells_by = "pseudo_time",
  cluster_rows = T,
  show_rownames = F,
  show_colnames = F,
  annotate_gene_clusters = F,
  annotation_row = row_ann,
  annotation_colors = annotation_colors
)
```


```{r, include = F, eval = F}
ggsave(filename = "embryo_devel_PCA_HDG.eps",device = "eps",plot = top_panel,path = "~/CorGI_figures/embryo_devel", width = 8,height = (4*8/6))
```



```{r}
plot_venn_diagram(gene_sets[1:3])
```

```{r}
plot_venn_diagram(gene_sets[c(1,5,9)])
names(gene_sets)
```

```{r, include = F}
ggsave(filename = "venn_diagram_HDG.eps",device = "eps",plot = venn_diagram,path = "~/CorGI_figures/embryo_devel",width = 4,height = 4)

```




# Brennecke highly variable genes

We evaluate the performance of the highly variable gene selection as proposed in [Brennecke et al., 2013](https://www.nature.com/articles/nmeth.2645) and implemented in the [M3Drop](https://bioconductor.org/packages/release/bioc/html/M3Drop.html) package in the `BrenneckeGetVariableGenes` function.

```{r}
# stop()
HVG_list_yan <-
  c(as.character(BrenneckeGetVariableGenes(normcounts(yan))[["Gene"]]),
    paste0("fake_gene",1:1000))

HVG_list_deng <-
  c(as.character(BrenneckeGetVariableGenes(counts(deng))[["Gene"]]),
    paste0("fake_gene",1:1000))

gene_sets <- get_compared_gene_sets(
  batch1_top_genes = HVG_list_yan,
  batch1_name = "HVG(Human)",
  batch2_top_genes = HVG_list_deng,
  batch2_name = "HVG(Mouse)",
  desired_size = length(corgi_gene_set)
)

lapply(gene_sets,
       function(gene_set){
         gene_set[!grepl("fake_gene",gene_set)]
       }) -> gene_sets
# 
# gene_sets <- list()
# gene_sets[["HVG(Human)"]] <- BrenneckeGetVariableGenes(normcounts(yan))[["Gene"]]
# gene_sets[["HVG(Mouse)"]] <- BrenneckeGetVariableGenes(counts(deng))[["Gene"]]
# gene_sets[["Union"]] <-
#   union(
#     gene_sets[["HVG(Human)"]],
#     gene_sets[["HVG(Mouse)"]]
#   )
# gene_sets[["Intersection"]] <-
#   intersect(
#     gene_sets[["HVG(Human)"]],
#     gene_sets[["HVG(Mouse)"]]
#   )
#   
set.seed(0)
gene_sets[["Random"]] <- sample(shared_genes,length(gene_sets[[1]]))
```

Reorder the gene sets
```{r}
gene_sets <- gene_sets[c(5,1:4)]
# make sure they are all size 200
lapply(gene_sets,length)

```



```{r}
embeddings_mds <- 
  lapply(
    X = gene_sets,
    FUN = function(gene_set) {
      spearman_rho_mds(combined[gene_set, ])
      # prcomp(t(combined[gene_set,]),rank. = 2)$x
    }
  )
```


```{r}
emb_name <- "MDS"
embeddings <- embeddings_mds


get_scatterplots(embeddings,
                 as.factor(batch),
                 cell_type) -> dr_plots

plot_grid(
  plot_grid(
    get_axes_legend(emb_name),
    get_shape_legend(batch, my_shape_palette), nrow = 2
  ),
  get_color_legend(cell_type, my_color_palette,ncol = 1),
  nrow = 1) -> legends

dr_plots[[length(dr_plots)+1]] <- legends

dr_plots[["nrow"]] <- 2

dr_plots[["labels"]] <- c("A","B","C","D","E","")

do.call(plot_grid,dr_plots) -> top_panel
top_panel
```


```{r, include = F}
ggsave(filename = "embryo_devel_PCA_HVG.eps",device = "eps",plot = top_panel,path = "~/CorGI_figures/embryo_devel", width = 8,height = (4*8/6))
```




```{r, include = F, eval = F}
# windows.options(width=10, height=10)

# setEPS()
# postscript("~/CorGI_figures/SVM.eps",width = 5,height = 5)
emb <- embeddings_pca[["Random"]]
emb <- emb
dat <- data.frame(emb,Batch = batch)
m <- e1071::svm(Batch~.,data=dat)


gx = seq(min(emb[,1]), max(emb[,1]), 0.1)
gy = seq(min(emb[,2]), max(emb[,2]), 0.1)
datagrid  = expand.grid(PC1 = gx, PC2 = gy)
 
# predict on it and add prediction to the grid
pred = predict(m, datagrid, decision.values = TRUE)
pred = attributes(pred)$decision.values
datagrid = cbind (datagrid, z = as.vector(pred))
# 
# plot(m,data=dat,svSymbol = "o", dataSymbol = "o")
# dev.off()
P = 
  ggplot(dat) + geom_point(aes( x=  PC1, y= PC2, shape = Batch, color = cell_type)) +
  scale_shape_manual(values = my_shape_palette)+
  scale_color_manual(values = my_color_palette)+
  # scale_colour_gradientn(colours=cols) + 
  # new_theme_empty + xlim(-7.5, 7.5) + ylim(-7.5, 7.5) +
  geom_contour( data=datagrid, aes(x=PC1, y=PC2, z=z), col = "black", size = 1, breaks=c(0), linetype =2) + theme_bw()+
  theme(
      plot.title = element_text(hjust = 0.5),
      axis.text = element_blank(),
      axis.ticks = element_blank(),
      axis.title = element_blank(),
      panel.grid = element_blank(),
      legend.position = "none"
    ) + ggtitle("SVM")

P

emb_name <- "PC"
embeddings <- embeddings_pca


get_scatterplots(embeddings,
                 as.factor(batch),
                 cell_type) -> dr_plots
dr_plots[[1]] <- P
plot_grid(
  plot_grid(
    get_axes_legend(emb_name),
    get_shape_legend(batch, my_shape_palette), nrow = 2
  ),
  get_color_legend(cell_type, my_color_palette,ncol = 1),
  nrow = 1) -> legends

dr_plots[[length(dr_plots)+1]] <- legends

dr_plots[["nrow"]] <- 2

dr_plots[["labels"]] <- c("A","B","C","D","E","")

do.call(plot_grid,dr_plots) -> top_panel
top_panel

ggsave(filename = "SVM.eps",device = "eps",plot = top_panel,path = "~/CorGI_figures/embryo_devel", width = 8,height = (4*8/6))
```





```{r, include = F, eval = F}
lapply(X = 1:6,function(i){
  union(select_top_corgi_genes(corgi_output_embyro_devel,
                               n = 100,SVG_use = 2,phase_use = i),
        select_top_corgi_genes(corgi_output_embyro_devel,
                               n = 100,SVG_use = 3,phase_use = i))
}) -> gene_sets
names(gene_sets) <- paste0("CorGI",1:6)

embeddings_pca <- 
  lapply(
    X = gene_sets,
    FUN = function(gene_set) {
      prcomp(t(combined[gene_set,]),rank. = 2)$x
    }
  )

emb_name <- "PC"
embeddings <- embeddings_pca


get_scatterplots(embeddings,
                 as.factor(batch),
                 cell_type) -> dr_plots

plot_grid(
  plot_grid(
    get_axes_legend(emb_name),
    get_shape_legend(batch, my_shape_palette), nrow = 2
  ),
  get_color_legend(cell_type, my_color_palette,ncol = 1),
  nrow = 1) -> legends

dr_plots[[length(dr_plots)+1]] <- legends

dr_plots[["nrow"]] <- 2

dr_plots[["labels"]] <- c("A","B","C","D","E","")

do.call(plot_grid,dr_plots) -> top_panel
top_panel
```
