<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Pre-implantation embryo development in human and mouse â€¢ corgi</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Pre-implantation embryo development in human and mouse">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">corgi</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/corgi.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/embryo-devel.html">Pre-implantation embryo development in human and mouse</a>
    </li>
    <li>
      <a href="../articles/liver-bud.html">Liver bud development</a>
    </li>
    <li>
      <a href="../articles/mammary-gland-devel.html">Mammary gland development</a>
    </li>
    <li>
      <a href="../articles/neurogenesis.html">Neurogenesis in the SVZ</a>
    </li>
    <li>
      <a href="../articles/toy-example.html">Synthetic data example</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Pre-implantation embryo development in human and mouse</h1>
                        <h4 class="author">Vignette Author</h4>
            
            <h4 class="date">2019-02-05</h4>
      
      
      <div class="hidden name"><code>embryo-devel.Rmd</code></div>

    </div>

    
    
<p>We illustrate the CorGI feature selection method by analyzing datasets on human and mouse pre-implantation embryo development.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(corgi)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(cowplot)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">#&gt; Warning: package 'cowplot' was built under R version 3.5.2</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co">#&gt; Attaching package: 'cowplot'</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">#&gt; The following object is masked from 'package:ggplot2':</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="co">#&gt;     ggsave</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scmap)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="co">#&gt; Warning: package 'scmap' was built under R version 3.5.2</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="co">#&gt; Creating a generic function for 'toJSON' from package 'jsonlite' in package 'googleVis'</span></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scran)</a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co">#&gt; Warning: package 'scran' was built under R version 3.5.2</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="co">#&gt; Loading required package: BiocParallel</span></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="co">#&gt; Warning: package 'BiocParallel' was built under R version 3.5.2</span></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="co">#&gt; Loading required package: SingleCellExperiment</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="co">#&gt; Warning: package 'SingleCellExperiment' was built under R version 3.5.2</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="co">#&gt; Loading required package: SummarizedExperiment</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="co">#&gt; Warning: package 'SummarizedExperiment' was built under R version 3.5.1</span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="co">#&gt; Loading required package: GenomicRanges</span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="co">#&gt; Warning: package 'GenomicRanges' was built under R version 3.5.1</span></a>
<a class="sourceLine" id="cb1-23" data-line-number="23"><span class="co">#&gt; Loading required package: stats4</span></a>
<a class="sourceLine" id="cb1-24" data-line-number="24"><span class="co">#&gt; Loading required package: BiocGenerics</span></a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="co">#&gt; Warning: package 'BiocGenerics' was built under R version 3.5.1</span></a>
<a class="sourceLine" id="cb1-26" data-line-number="26"><span class="co">#&gt; Loading required package: parallel</span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></a>
<a class="sourceLine" id="cb1-29" data-line-number="29"><span class="co">#&gt; The following objects are masked from 'package:parallel':</span></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-31" data-line-number="31"><span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span></a>
<a class="sourceLine" id="cb1-32" data-line-number="32"><span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span></a>
<a class="sourceLine" id="cb1-33" data-line-number="33"><span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span></a>
<a class="sourceLine" id="cb1-34" data-line-number="34"><span class="co">#&gt; The following objects are masked from 'package:stats':</span></a>
<a class="sourceLine" id="cb1-35" data-line-number="35"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-36" data-line-number="36"><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></a>
<a class="sourceLine" id="cb1-37" data-line-number="37"><span class="co">#&gt; The following objects are masked from 'package:base':</span></a>
<a class="sourceLine" id="cb1-38" data-line-number="38"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-39" data-line-number="39"><span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind,</span></a>
<a class="sourceLine" id="cb1-40" data-line-number="40"><span class="co">#&gt;     colMeans, colnames, colSums, dirname, do.call, duplicated,</span></a>
<a class="sourceLine" id="cb1-41" data-line-number="41"><span class="co">#&gt;     eval, evalq, Filter, Find, get, grep, grepl, intersect,</span></a>
<a class="sourceLine" id="cb1-42" data-line-number="42"><span class="co">#&gt;     is.unsorted, lapply, lengths, Map, mapply, match, mget, order,</span></a>
<a class="sourceLine" id="cb1-43" data-line-number="43"><span class="co">#&gt;     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,</span></a>
<a class="sourceLine" id="cb1-44" data-line-number="44"><span class="co">#&gt;     Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,</span></a>
<a class="sourceLine" id="cb1-45" data-line-number="45"><span class="co">#&gt;     table, tapply, union, unique, unsplit, which, which.max,</span></a>
<a class="sourceLine" id="cb1-46" data-line-number="46"><span class="co">#&gt;     which.min</span></a>
<a class="sourceLine" id="cb1-47" data-line-number="47"><span class="co">#&gt; Loading required package: S4Vectors</span></a>
<a class="sourceLine" id="cb1-48" data-line-number="48"><span class="co">#&gt; Warning: package 'S4Vectors' was built under R version 3.5.1</span></a>
<a class="sourceLine" id="cb1-49" data-line-number="49"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-50" data-line-number="50"><span class="co">#&gt; Attaching package: 'S4Vectors'</span></a>
<a class="sourceLine" id="cb1-51" data-line-number="51"><span class="co">#&gt; The following object is masked from 'package:base':</span></a>
<a class="sourceLine" id="cb1-52" data-line-number="52"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-53" data-line-number="53"><span class="co">#&gt;     expand.grid</span></a>
<a class="sourceLine" id="cb1-54" data-line-number="54"><span class="co">#&gt; Loading required package: IRanges</span></a>
<a class="sourceLine" id="cb1-55" data-line-number="55"><span class="co">#&gt; Warning: package 'IRanges' was built under R version 3.5.1</span></a>
<a class="sourceLine" id="cb1-56" data-line-number="56"><span class="co">#&gt; Loading required package: GenomeInfoDb</span></a>
<a class="sourceLine" id="cb1-57" data-line-number="57"><span class="co">#&gt; Warning: package 'GenomeInfoDb' was built under R version 3.5.1</span></a>
<a class="sourceLine" id="cb1-58" data-line-number="58"><span class="co">#&gt; Loading required package: Biobase</span></a>
<a class="sourceLine" id="cb1-59" data-line-number="59"><span class="co">#&gt; Warning: package 'Biobase' was built under R version 3.5.1</span></a>
<a class="sourceLine" id="cb1-60" data-line-number="60"><span class="co">#&gt; Welcome to Bioconductor</span></a>
<a class="sourceLine" id="cb1-61" data-line-number="61"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-62" data-line-number="62"><span class="co">#&gt;     Vignettes contain introductory material; view with</span></a>
<a class="sourceLine" id="cb1-63" data-line-number="63"><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></a>
<a class="sourceLine" id="cb1-64" data-line-number="64"><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></a>
<a class="sourceLine" id="cb1-65" data-line-number="65"><span class="co">#&gt; Loading required package: DelayedArray</span></a>
<a class="sourceLine" id="cb1-66" data-line-number="66"><span class="co">#&gt; Warning: package 'DelayedArray' was built under R version 3.5.1</span></a>
<a class="sourceLine" id="cb1-67" data-line-number="67"><span class="co">#&gt; Loading required package: matrixStats</span></a>
<a class="sourceLine" id="cb1-68" data-line-number="68"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-69" data-line-number="69"><span class="co">#&gt; Attaching package: 'matrixStats'</span></a>
<a class="sourceLine" id="cb1-70" data-line-number="70"><span class="co">#&gt; The following objects are masked from 'package:Biobase':</span></a>
<a class="sourceLine" id="cb1-71" data-line-number="71"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-72" data-line-number="72"><span class="co">#&gt;     anyMissing, rowMedians</span></a>
<a class="sourceLine" id="cb1-73" data-line-number="73"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-74" data-line-number="74"><span class="co">#&gt; Attaching package: 'DelayedArray'</span></a>
<a class="sourceLine" id="cb1-75" data-line-number="75"><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></a>
<a class="sourceLine" id="cb1-76" data-line-number="76"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-77" data-line-number="77"><span class="co">#&gt;     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span></a>
<a class="sourceLine" id="cb1-78" data-line-number="78"><span class="co">#&gt; The following objects are masked from 'package:base':</span></a>
<a class="sourceLine" id="cb1-79" data-line-number="79"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-80" data-line-number="80"><span class="co">#&gt;     aperm, apply</span></a>
<a class="sourceLine" id="cb1-81" data-line-number="81"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(M3Drop)</a>
<a class="sourceLine" id="cb1-82" data-line-number="82"><span class="co">#&gt; Warning: package 'M3Drop' was built under R version 3.5.2</span></a>
<a class="sourceLine" id="cb1-83" data-line-number="83"><span class="co">#&gt; Loading required package: numDeriv</span></a></code></pre></div>
<div id="preprocessing-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#preprocessing-the-data" class="anchor"></a>Preprocessing the data</h1>
<p>We download the data from the <a href="https://hemberg-lab.github.io/scRNA.seq.datasets/">Hemberg Lab website</a>. The human dataset is from <a href="http://dx.doi.org/10.1038/nsmb.2660">Yan et al 2013</a> while the mouse dataset is from <a href="http://dx.doi.org/10.1126/science.1245316">Deng et al 2014</a>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">yan &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/readRDS">readRDS</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/connections">url</a></span>(<span class="st">"https://scrnaseq-public-datasets.s3.amazonaws.com/scater-objects/yan.rds"</span>))</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">deng &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/readRDS">readRDS</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/connections">url</a></span>(<span class="st">"https://scrnaseq-public-datasets.s3.amazonaws.com/scater-objects/deng-reads.rds"</span>))</a></code></pre></div>
<p>Get genes common to both datasets</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(yan) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/chartr">toupper</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(yan))</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(deng) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/chartr">toupper</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(deng))</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">shared_genes &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sets">intersect</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(yan),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(deng))</a></code></pre></div>
<p>Get rid of the spike-in genes</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">shared_genes &lt;-<span class="st"> </span>shared_genes[<span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">"ERCC"</span>,shared_genes)]</a></code></pre></div>
<p>Subset the data</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">yan &lt;-<span class="st"> </span>yan[shared_genes,]</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">deng &lt;-<span class="st"> </span>deng[shared_genes,]</a></code></pre></div>
</div>
<div id="run-corgi-feature-selection" class="section level1">
<h1 class="hasAnchor">
<a href="#run-corgi-feature-selection" class="anchor"></a>Run CorGI feature selection</h1>
<p>Normalization is not necessary for running CorGI</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">corgi_output_embyro_devel &lt;-<span class="st"> </span><span class="kw"><a href="../reference/run_corgi.html">run_corgi</a></span>(<span class="kw">logcounts</span>(yan),<span class="kw">logcounts</span>(deng))</a></code></pre></div>
<p>Select top CorGI genes</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">corgi_gene_set &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sets">union</a></span>(<span class="kw"><a href="../reference/select_top_corgi_genes.html">select_top_corgi_genes</a></span>(corgi_output_embyro_devel,<span class="dt">n =</span> <span class="dv">100</span>,<span class="dt">SVG_use =</span> <span class="dv">2</span>),</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">        <span class="kw"><a href="../reference/select_top_corgi_genes.html">select_top_corgi_genes</a></span>(corgi_output_embyro_devel,<span class="dt">n =</span> <span class="dv">100</span>,<span class="dt">SVG_use =</span> <span class="dv">3</span>))</a></code></pre></div>
</div>
<div id="visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h1>
<p>Normalization</p>
<p>Metadata for the combined datasets</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">cell_type &lt;-<span class="st"> </span>forcats<span class="op">::</span><span class="kw"><a href="https://forcats.tidyverse.org/reference/fct_c.html">fct_c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">as.factor</a></span>(yan<span class="op">$</span>cell_type1),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">as.factor</a></span>(deng<span class="op">$</span>cell_type1))</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">cell_type &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">factor</a></span>(cell_type, <span class="dt">levels =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"zygote"</span>,<span class="st">"2cell"</span>,<span class="st">"4cell"</span>,<span class="st">"8cell"</span>,<span class="st">"16cell"</span>,<span class="st">"blast"</span>))</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">batch &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="st">"Human"</span>,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(yan)),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="st">"Mouse"</span>,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(deng)))</a></code></pre></div>
<p>Wrapper around the qplot <code><a href="https://ggplot2.tidyverse.org/reference/qplot.html">ggplot2::qplot</a></code> function with our color scheme</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">my_color_palette &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"#000000"</span>, <span class="st">"#E69F00"</span>, <span class="st">"#56B4E9"</span>, <span class="st">"#009E73"</span>, <span class="st">"#F0E442"</span>, <span class="st">"#0072B2"</span>, <span class="st">"#D55E00"</span>, <span class="st">"#CC79A7"</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">my_shape_palette &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">16</span>,<span class="dv">1</span>)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">qplot &lt;-<span class="st"> </span><span class="cf">function</span>(...){</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">  ggplot2<span class="op">::</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/qplot.html">qplot</a></span>(...) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span>(<span class="dt">values =</span> my_color_palette) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_shape_manual</a></span>(<span class="dt">values =</span> my_shape_palette)</a>
<a class="sourceLine" id="cb9-7" data-line-number="7">}</a></code></pre></div>
<div id="principal-component-analysis-pca" class="section level2">
<h2 class="hasAnchor">
<a href="#principal-component-analysis-pca" class="anchor"></a>Principal component analysis (PCA)</h2>
<p>To use PCA, we first need to adjust the read-depth to be comparable across the two datasets</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scran)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">out &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/scran/topics/multiBatchNorm">multiBatchNorm</a></span>(yan,deng,<span class="dt">assay.type =</span> <span class="st">"logcounts"</span>)</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">#&gt; Warning in .batch_rescaler(batches, subset.row = nonspike.subset,</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt; exprs_values = assay.type, : no size factors in batch 1, using sum of</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#&gt; counts instead</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#&gt; Warning in .batch_rescaler(batches, subset.row = nonspike.subset,</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt; exprs_values = assay.type, : no size factors in batch 2, using sum of</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt; counts instead</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="co">#&gt; Warning in .get_all_sf_sets(object): spike-in set 'ERCC' should have its</span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="co">#&gt; own size factors</span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="co">#&gt; Warning in .get_all_sf_sets(object): spike-in set 'ERCC' should have its</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="co">#&gt; own size factors</span></a>
<a class="sourceLine" id="cb10-14" data-line-number="14">yan &lt;-<span class="st"> </span>out[[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb10-15" data-line-number="15">deng &lt;-<span class="st"> </span>out[[<span class="dv">2</span>]]</a>
<a class="sourceLine" id="cb10-16" data-line-number="16">combined &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">cbind</a></span>(<span class="kw">logcounts</span>(yan),<span class="kw">logcounts</span>(deng))</a></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">HDG_ranking &lt;-<span class="st"> </span><span class="cf">function</span>(sce){</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">  <span class="kw">rowData</span>(sce)<span class="op">$</span>feature_symbol &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">  sce &lt;-<span class="st"> </span>scmap<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scmap/topics/selectFeatures">selectFeatures</a></span>(sce)</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/function">return</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce)[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/order">order</a></span>(<span class="kw">rowData</span>(sce)[[<span class="st">"scmap_scores"</span>]],<span class="dt">decreasing =</span> T,<span class="dt">na.last =</span> T)])</a>
<a class="sourceLine" id="cb11-5" data-line-number="5">}</a>
<a class="sourceLine" id="cb11-6" data-line-number="6"></a>
<a class="sourceLine" id="cb11-7" data-line-number="7">gene_sets &lt;-<span class="st"> </span><span class="kw">get_compared_gene_sets</span>(</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">  <span class="dt">batch1_top_genes =</span> <span class="kw">HDG_ranking</span>(yan),</a>
<a class="sourceLine" id="cb11-9" data-line-number="9">  <span class="dt">batch1_name =</span> <span class="st">"HDG(Human)"</span>,</a>
<a class="sourceLine" id="cb11-10" data-line-number="10">  <span class="dt">batch2_top_genes =</span> <span class="kw">HDG_ranking</span>(deng),</a>
<a class="sourceLine" id="cb11-11" data-line-number="11">  <span class="dt">batch2_name =</span> <span class="st">"HDG(Mouse)"</span>,</a>
<a class="sourceLine" id="cb11-12" data-line-number="12">  <span class="dt">desired_size =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(corgi_gene_set)</a>
<a class="sourceLine" id="cb11-13" data-line-number="13">)</a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="co">#&gt; Warning in linearModel(object, n_features): Your object does not contain</span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="co">#&gt; counts() slot. Dropouts were calculated using logcounts() slot...</span></a>
<a class="sourceLine" id="cb11-16" data-line-number="16"></a>
<a class="sourceLine" id="cb11-17" data-line-number="17">gene_sets[[<span class="st">"CorGI"</span>]] &lt;-<span class="st"> </span>corgi_gene_set</a>
<a class="sourceLine" id="cb11-18" data-line-number="18"></a>
<a class="sourceLine" id="cb11-19" data-line-number="19"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(gene_sets)</a>
<a class="sourceLine" id="cb11-20" data-line-number="20"><span class="co">#&gt; [1] "HDG(Human)"   "HDG(Mouse)"   "Union"        "Intersection"</span></a>
<a class="sourceLine" id="cb11-21" data-line-number="21"><span class="co">#&gt; [5] "CorGI"</span></a></code></pre></div>
<p>Reorder the gene sets</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">gene_sets &lt;-<span class="st"> </span>gene_sets[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>)]</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(gene_sets, length)</a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co">#&gt; $CorGI</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="co">#&gt; [1] 200</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="co">#&gt; $`HDG(Human)`</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="co">#&gt; [1] 200</span></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="co">#&gt; $`HDG(Mouse)`</span></a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="co">#&gt; [1] 200</span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb12-12" data-line-number="12"><span class="co">#&gt; $Union</span></a>
<a class="sourceLine" id="cb12-13" data-line-number="13"><span class="co">#&gt; [1] 200</span></a>
<a class="sourceLine" id="cb12-14" data-line-number="14"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb12-15" data-line-number="15"><span class="co">#&gt; $Intersection</span></a>
<a class="sourceLine" id="cb12-16" data-line-number="16"><span class="co">#&gt; [1] 200</span></a></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">embeddings_pca &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">    <span class="dt">X =</span> gene_sets,</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">    <span class="dt">FUN =</span> <span class="cf">function</span>(gene_set) {</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">      <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/prcomp">prcomp</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(combined[gene_set,]),<span class="dt">rank. =</span> <span class="dv">2</span>)<span class="op">$</span>x</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">    }</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">  )</a></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">emb_name &lt;-<span class="st"> "PC"</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2">embeddings &lt;-<span class="st"> </span>embeddings_pca</a>
<a class="sourceLine" id="cb14-3" data-line-number="3"></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="kw">get_scatterplots</span>(embeddings,</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">                 <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">as.factor</a></span>(batch),</a>
<a class="sourceLine" id="cb14-7" data-line-number="7">                 cell_type) -&gt;<span class="st"> </span>dr_plots</a>
<a class="sourceLine" id="cb14-8" data-line-number="8"></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb14-10" data-line-number="10">  <span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb14-11" data-line-number="11">    <span class="kw"><a href="../reference/get_axes_legend.html">get_axes_legend</a></span>(emb_name),</a>
<a class="sourceLine" id="cb14-12" data-line-number="12">    <span class="kw">get_shape_legend</span>(batch, my_shape_palette), <span class="dt">nrow =</span> <span class="dv">2</span></a>
<a class="sourceLine" id="cb14-13" data-line-number="13">  ),</a>
<a class="sourceLine" id="cb14-14" data-line-number="14">  <span class="kw">get_color_legend</span>(cell_type, my_color_palette,<span class="dt">ncol =</span> <span class="dv">1</span>),</a>
<a class="sourceLine" id="cb14-15" data-line-number="15">  <span class="dt">nrow =</span> <span class="dv">1</span>) -&gt;<span class="st"> </span>legends</a>
<a class="sourceLine" id="cb14-16" data-line-number="16"></a>
<a class="sourceLine" id="cb14-17" data-line-number="17">dr_plots[[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(dr_plots)<span class="op">+</span><span class="dv">1</span>]] &lt;-<span class="st"> </span>legends</a>
<a class="sourceLine" id="cb14-18" data-line-number="18"></a>
<a class="sourceLine" id="cb14-19" data-line-number="19">dr_plots[[<span class="st">"nrow"</span>]] &lt;-<span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb14-20" data-line-number="20"></a>
<a class="sourceLine" id="cb14-21" data-line-number="21">dr_plots[[<span class="st">"labels"</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"A"</span>,<span class="st">"B"</span>,<span class="st">"C"</span>,<span class="st">"D"</span>,<span class="st">"E"</span>,<span class="st">""</span>)</a>
<a class="sourceLine" id="cb14-22" data-line-number="22"></a>
<a class="sourceLine" id="cb14-23" data-line-number="23"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/do.call">do.call</a></span>(plot_grid,dr_plots) -&gt;<span class="st"> </span>top_panel</a>
<a class="sourceLine" id="cb14-24" data-line-number="24">top_panel</a></code></pre></div>
<p><img src="embryo-devel_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span>(<span class="dt">filename =</span> <span class="st">"embryo_devel_PCA_HDG.eps"</span>,<span class="dt">device =</span> <span class="st">"eps"</span>,<span class="dt">plot =</span> top_panel,<span class="dt">path =</span> <span class="st">"~/CorGI_figures/embryo_devel"</span>,<span class="dt">width =</span> <span class="dv">6</span>,<span class="dt">height =</span> <span class="dv">4</span>)</a></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggforce)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(limma)</a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="co">#&gt; Warning: package 'limma' was built under R version 3.5.1</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="co">#&gt; Attaching package: 'limma'</span></a>
<a class="sourceLine" id="cb16-6" data-line-number="6"><span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span></a>
<a class="sourceLine" id="cb16-7" data-line-number="7"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-8" data-line-number="8"><span class="co">#&gt;     plotMA</span></a>
<a class="sourceLine" id="cb16-9" data-line-number="9"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb16-10" data-line-number="10"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-11" data-line-number="11"><span class="co">#&gt; Attaching package: 'dplyr'</span></a>
<a class="sourceLine" id="cb16-12" data-line-number="12"><span class="co">#&gt; The following object is masked from 'package:matrixStats':</span></a>
<a class="sourceLine" id="cb16-13" data-line-number="13"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-14" data-line-number="14"><span class="co">#&gt;     count</span></a>
<a class="sourceLine" id="cb16-15" data-line-number="15"><span class="co">#&gt; The following object is masked from 'package:Biobase':</span></a>
<a class="sourceLine" id="cb16-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-17" data-line-number="17"><span class="co">#&gt;     combine</span></a>
<a class="sourceLine" id="cb16-18" data-line-number="18"><span class="co">#&gt; The following objects are masked from 'package:GenomicRanges':</span></a>
<a class="sourceLine" id="cb16-19" data-line-number="19"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-20" data-line-number="20"><span class="co">#&gt;     intersect, setdiff, union</span></a>
<a class="sourceLine" id="cb16-21" data-line-number="21"><span class="co">#&gt; The following object is masked from 'package:GenomeInfoDb':</span></a>
<a class="sourceLine" id="cb16-22" data-line-number="22"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-23" data-line-number="23"><span class="co">#&gt;     intersect</span></a>
<a class="sourceLine" id="cb16-24" data-line-number="24"><span class="co">#&gt; The following objects are masked from 'package:IRanges':</span></a>
<a class="sourceLine" id="cb16-25" data-line-number="25"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-26" data-line-number="26"><span class="co">#&gt;     collapse, desc, intersect, setdiff, slice, union</span></a>
<a class="sourceLine" id="cb16-27" data-line-number="27"><span class="co">#&gt; The following objects are masked from 'package:S4Vectors':</span></a>
<a class="sourceLine" id="cb16-28" data-line-number="28"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-29" data-line-number="29"><span class="co">#&gt;     first, intersect, rename, setdiff, setequal, union</span></a>
<a class="sourceLine" id="cb16-30" data-line-number="30"><span class="co">#&gt; The following objects are masked from 'package:BiocGenerics':</span></a>
<a class="sourceLine" id="cb16-31" data-line-number="31"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-32" data-line-number="32"><span class="co">#&gt;     combine, intersect, setdiff, union</span></a>
<a class="sourceLine" id="cb16-33" data-line-number="33"><span class="co">#&gt; The following objects are masked from 'package:stats':</span></a>
<a class="sourceLine" id="cb16-34" data-line-number="34"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-35" data-line-number="35"><span class="co">#&gt;     filter, lag</span></a>
<a class="sourceLine" id="cb16-36" data-line-number="36"><span class="co">#&gt; The following objects are masked from 'package:base':</span></a>
<a class="sourceLine" id="cb16-37" data-line-number="37"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-38" data-line-number="38"><span class="co">#&gt;     intersect, setdiff, setequal, union</span></a>
<a class="sourceLine" id="cb16-39" data-line-number="39">df.venn &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">x =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="fl">0.866</span>, <span class="fl">-0.866</span>),</a>
<a class="sourceLine" id="cb16-40" data-line-number="40">                      <span class="dt">y =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">-0.5</span>, <span class="fl">-0.5</span>),</a>
<a class="sourceLine" id="cb16-41" data-line-number="41">                      <span class="dt">labels =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'CorGI'</span>, <span class="st">'HDG(Human)'</span>, <span class="st">'HDG(Mouse)'</span>))</a>
<a class="sourceLine" id="cb16-42" data-line-number="42"></a>
<a class="sourceLine" id="cb16-43" data-line-number="43">all_genes &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/funprog">Reduce</a></span>(union, gene_sets[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>])</a>
<a class="sourceLine" id="cb16-44" data-line-number="44"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(gene_sets)[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a>
<a class="sourceLine" id="cb16-45" data-line-number="45"><span class="co">#&gt; [1] "CorGI"      "HDG(Human)" "HDG(Mouse)"</span></a>
<a class="sourceLine" id="cb16-46" data-line-number="46">mydata &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb16-47" data-line-number="47"><span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">CorGI =</span> all_genes <span class="op">%in%</span><span class="st"> </span>gene_sets[[<span class="st">"CorGI"</span>]],</a>
<a class="sourceLine" id="cb16-48" data-line-number="48">             <span class="st">`</span><span class="dt">HDG(Human)</span><span class="st">`</span> =<span class="st"> </span>all_genes <span class="op">%in%</span><span class="st"> </span>gene_sets[[<span class="st">"HDG(Human)"</span>]],</a>
<a class="sourceLine" id="cb16-49" data-line-number="49">             <span class="st">`</span><span class="dt">HDG(Mouse)</span><span class="st">`</span> =<span class="st"> </span>all_genes <span class="op">%in%</span><span class="st"> </span>gene_sets[[<span class="st">"HDG(Mouse)"</span>]])</a>
<a class="sourceLine" id="cb16-50" data-line-number="50"></a>
<a class="sourceLine" id="cb16-51" data-line-number="51">vdc &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/limma/topics/venn">vennCounts</a></span>(mydata)</a>
<a class="sourceLine" id="cb16-52" data-line-number="52"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/class">class</a></span>(vdc) &lt;-<span class="st"> 'matrix'</span></a>
<a class="sourceLine" id="cb16-53" data-line-number="53">df.vdc &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(vdc)[<span class="op">-</span><span class="dv">1</span>,] <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-54" data-line-number="54"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">x =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="fl">1.2</span>, <span class="fl">0.8</span>, <span class="fl">-1.2</span>, <span class="fl">-0.8</span>, <span class="dv">0</span>, <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb16-55" data-line-number="55">         <span class="dt">y =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="fl">1.2</span>, <span class="fl">-0.6</span>, <span class="fl">0.5</span>, <span class="fl">-0.6</span>, <span class="fl">0.5</span>, <span class="dv">-1</span>, <span class="dv">0</span>))</a>
<a class="sourceLine" id="cb16-56" data-line-number="56"></a>
<a class="sourceLine" id="cb16-57" data-line-number="57">df.gene_set_names &lt;-</a>
<a class="sourceLine" id="cb16-58" data-line-number="58"><span class="st">  </span>df.vdc[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>),<span class="dv">4</span><span class="op">:</span><span class="dv">6</span>]</a>
<a class="sourceLine" id="cb16-59" data-line-number="59">df.gene_set_names</a>
<a class="sourceLine" id="cb16-60" data-line-number="60"><span class="co">#&gt;   Counts    x    y</span></a>
<a class="sourceLine" id="cb16-61" data-line-number="61"><span class="co">#&gt; 1    161  0.0  1.2</span></a>
<a class="sourceLine" id="cb16-62" data-line-number="62"><span class="co">#&gt; 2    149  1.2 -0.6</span></a>
<a class="sourceLine" id="cb16-63" data-line-number="63"><span class="co">#&gt; 4    133 -1.2 -0.6</span></a>
<a class="sourceLine" id="cb16-64" data-line-number="64"></a>
<a class="sourceLine" id="cb16-65" data-line-number="65">df.gene_set_names<span class="op">$</span>Name &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"HDG(Mouse)"</span>,<span class="st">"HDG(Human)"</span>,<span class="st">"CorGI"</span>)</a>
<a class="sourceLine" id="cb16-66" data-line-number="66">df.gene_set_names<span class="op">$</span>y[<span class="dv">1</span>] &lt;-<span class="st"> </span>df.gene_set_names<span class="op">$</span>y[<span class="dv">1</span>] <span class="op">+</span><span class="st"> </span><span class="fl">0.5</span></a>
<a class="sourceLine" id="cb16-67" data-line-number="67">df.gene_set_names<span class="op">$</span>y[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>] &lt;-<span class="st"> </span>df.gene_set_names<span class="op">$</span>y[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>] <span class="fl">-0.5</span></a>
<a class="sourceLine" id="cb16-68" data-line-number="68">df.gene_set_names<span class="op">$</span>x[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>] &lt;-<span class="st"> </span>df.gene_set_names<span class="op">$</span>x[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>]<span class="op">*</span><span class="fl">1.25</span></a>
<a class="sourceLine" id="cb16-69" data-line-number="69"></a>
<a class="sourceLine" id="cb16-70" data-line-number="70"></a>
<a class="sourceLine" id="cb16-71" data-line-number="71"><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(df.venn) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-72" data-line-number="72"><span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggforce/topics/geom_circle">geom_circle</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x0 =</span> x, <span class="dt">y0 =</span> y, <span class="dt">r =</span> <span class="fl">1.5</span>, <span class="dt">fill =</span> labels), <span class="dt">alpha =</span> <span class="fl">.3</span>, <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">colour =</span> <span class="st">'grey'</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-73" data-line-number="73"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html">coord_fixed</a></span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb16-74" data-line-number="74"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_void</a></span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb16-75" data-line-number="75"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">legend.position =</span> <span class="st">'none'</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-76" data-line-number="76"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span>(<span class="dt">values =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'cornflowerblue'</span>, <span class="st">'firebrick'</span>,  <span class="st">'gold'</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-77" data-line-number="77"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_colour_manual</a></span>(<span class="dt">values =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'cornflowerblue'</span>, <span class="st">'firebrick'</span>, <span class="st">'gold'</span>), <span class="dt">guide =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-78" data-line-number="78"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">fill =</span> <span class="ot">NULL</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-79" data-line-number="79"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/annotate.html">annotate</a></span>(<span class="st">"text"</span>, <span class="dt">x =</span> df.vdc<span class="op">$</span>x, <span class="dt">y =</span> df.vdc<span class="op">$</span>y, <span class="dt">label =</span> df.vdc<span class="op">$</span>Counts, <span class="dt">size =</span> <span class="dv">5</span>)<span class="op">+</span></a>
<a class="sourceLine" id="cb16-80" data-line-number="80"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/annotate.html">annotate</a></span>(<span class="st">"text"</span>, <span class="dt">x =</span> df.gene_set_names<span class="op">$</span>x, <span class="dt">y =</span> df.gene_set_names<span class="op">$</span>y, <span class="dt">label =</span> df.gene_set_names<span class="op">$</span>Name) -&gt;</a>
<a class="sourceLine" id="cb16-81" data-line-number="81"><span class="st">  </span>venn_diagram</a>
<a class="sourceLine" id="cb16-82" data-line-number="82"></a>
<a class="sourceLine" id="cb16-83" data-line-number="83">venn_diagram</a></code></pre></div>
<p><img src="embryo-devel_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span>(<span class="dt">filename =</span> <span class="st">"venn_diagram_HDG.eps"</span>,<span class="dt">device =</span> <span class="st">"eps"</span>,<span class="dt">plot =</span> venn_diagram,<span class="dt">path =</span> <span class="st">"~/CorGI_figures/embryo_devel"</span>,<span class="dt">width =</span> <span class="dv">4</span>,<span class="dt">height =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="co">#&gt; Warning in grid.Call.graphics(C_polygon, x$x, x$y, index): semi-</span></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="co">#&gt; transparency is not supported on this device: reported only once per page</span></a></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="co"># plts &lt;- list()</span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(gene_sets)[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>],<span class="cf">function</span>(gs_name){</a>
<a class="sourceLine" id="cb18-3" data-line-number="3">  gene_set &lt;-<span class="st"> </span>gene_sets[[gs_name]]</a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/limma/topics/cbind">cbind</a></span>(</a>
<a class="sourceLine" id="cb18-5" data-line-number="5">  <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/cor">cor</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(combined[gene_set,</a>
<a class="sourceLine" id="cb18-6" data-line-number="6">                 batch <span class="op">==</span><span class="st"> "Mouse"</span>]),</a>
<a class="sourceLine" id="cb18-7" data-line-number="7">      <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(cell_type[batch <span class="op">==</span><span class="st"> "Mouse"</span>]),</a>
<a class="sourceLine" id="cb18-8" data-line-number="8">      <span class="dt">method =</span> <span class="st">"spearman"</span>),</a>
<a class="sourceLine" id="cb18-9" data-line-number="9">  <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/cor">cor</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(combined[gene_set,</a>
<a class="sourceLine" id="cb18-10" data-line-number="10">                 batch <span class="op">==</span><span class="st"> "Human"</span>]),</a>
<a class="sourceLine" id="cb18-11" data-line-number="11">      <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(cell_type[batch <span class="op">==</span><span class="st"> "Human"</span>]),</a>
<a class="sourceLine" id="cb18-12" data-line-number="12">      <span class="dt">method =</span> <span class="st">"spearman"</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb18-13" data-line-number="13"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">stringsAsFactors =</span> F) -&gt;<span class="st"> </span></a>
<a class="sourceLine" id="cb18-14" data-line-number="14"><span class="st">    </span>corr_wrt_devel</a>
<a class="sourceLine" id="cb18-15" data-line-number="15">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(corr_wrt_devel) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Mouse"</span>,<span class="st">"Human"</span>)</a>
<a class="sourceLine" id="cb18-16" data-line-number="16">  corr_wrt_devel<span class="op">$</span>Gene_set &lt;-<span class="st"> </span>gs_name</a>
<a class="sourceLine" id="cb18-17" data-line-number="17">  corr_wrt_devel</a>
<a class="sourceLine" id="cb18-18" data-line-number="18">}) -&gt;<span class="st"> </span>dfs</a>
<a class="sourceLine" id="cb18-19" data-line-number="19"><span class="co">#&gt; Warning in cor(t(combined[gene_set, batch == "Mouse"]),</span></a>
<a class="sourceLine" id="cb18-20" data-line-number="20"><span class="co">#&gt; as.numeric(cell_type[batch == : the standard deviation is zero</span></a>
<a class="sourceLine" id="cb18-21" data-line-number="21"><span class="co">#&gt; Warning in cor(t(combined[gene_set, batch == "Human"]),</span></a>
<a class="sourceLine" id="cb18-22" data-line-number="22"><span class="co">#&gt; as.numeric(cell_type[batch == : the standard deviation is zero</span></a>
<a class="sourceLine" id="cb18-23" data-line-number="23"></a>
<a class="sourceLine" id="cb18-24" data-line-number="24">reshape2<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/reshape2/topics/melt">melt</a></span>(<span class="kw"><a href="https://dplyr.tidyverse.org/reference/bind.html">rbind_list</a></span>(dfs)) -&gt;<span class="st"> </span>df</a>
<a class="sourceLine" id="cb18-25" data-line-number="25"><span class="co">#&gt; Warning: 'rbind_list' is deprecated.</span></a>
<a class="sourceLine" id="cb18-26" data-line-number="26"><span class="co">#&gt; Use 'bind_rows()' instead.</span></a>
<a class="sourceLine" id="cb18-27" data-line-number="27"><span class="co">#&gt; See help("Deprecated")</span></a>
<a class="sourceLine" id="cb18-28" data-line-number="28"><span class="co">#&gt; Using Gene_set as id variables</span></a>
<a class="sourceLine" id="cb18-29" data-line-number="29"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(df) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Gene_set"</span>, <span class="st">"Species"</span>,<span class="st">"Corr"</span>)</a>
<a class="sourceLine" id="cb18-30" data-line-number="30"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(df)</a>
<a class="sourceLine" id="cb18-31" data-line-number="31"><span class="co">#&gt;   Gene_set Species       Corr</span></a>
<a class="sourceLine" id="cb18-32" data-line-number="32"><span class="co">#&gt; 1    CorGI   Mouse -0.6449947</span></a>
<a class="sourceLine" id="cb18-33" data-line-number="33"><span class="co">#&gt; 2    CorGI   Mouse -0.6686111</span></a>
<a class="sourceLine" id="cb18-34" data-line-number="34"><span class="co">#&gt; 3    CorGI   Mouse -0.7282891</span></a>
<a class="sourceLine" id="cb18-35" data-line-number="35"><span class="co">#&gt; 4    CorGI   Mouse -0.6608965</span></a>
<a class="sourceLine" id="cb18-36" data-line-number="36"><span class="co">#&gt; 5    CorGI   Mouse -0.6437593</span></a>
<a class="sourceLine" id="cb18-37" data-line-number="37"><span class="co">#&gt; 6    CorGI   Mouse -0.5734754</span></a>
<a class="sourceLine" id="cb18-38" data-line-number="38">df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb18-39" data-line-number="39"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> Gene_set, <span class="dt">y =</span> Corr, <span class="dt">fill=</span>Species)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb18-40" data-line-number="40"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_violin.html">geom_violin</a></span>(<span class="dt">adjust =</span> <span class="fl">.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb18-41" data-line-number="41"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">legend.position =</span> <span class="st">"bottom"</span>,<span class="dt">axis.title.x =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>()) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"Correlation with</span><span class="ch">\n</span><span class="st"> developmental stage"</span>)-&gt;<span class="st"> </span>corr_violin</a>
<a class="sourceLine" id="cb18-42" data-line-number="42"><span class="co"># reshape2::melt(rbind_list(dfs[[1]])) </span></a></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(venn_diagram,corr_violin, <span class="dt">rel_heights =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">3</span>)) -&gt;<span class="st"> </span>bottom_panel</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="co">#&gt; Warning: Removed 2 rows containing non-finite values (stat_ydensity).</span></a></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="co"># plts &lt;- list()</span></a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(gene_sets,<span class="cf">function</span>(gene_set){</a>
<a class="sourceLine" id="cb20-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/limma/topics/cbind">cbind</a></span>(</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">  <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/cor">cor</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(combined[gene_set,</a>
<a class="sourceLine" id="cb20-5" data-line-number="5">                 batch <span class="op">==</span><span class="st"> "Mouse"</span>]),</a>
<a class="sourceLine" id="cb20-6" data-line-number="6">      <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(cell_type[batch <span class="op">==</span><span class="st"> "Mouse"</span>]),</a>
<a class="sourceLine" id="cb20-7" data-line-number="7">      <span class="dt">method =</span> <span class="st">"spearman"</span>),</a>
<a class="sourceLine" id="cb20-8" data-line-number="8">  <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/cor">cor</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(combined[gene_set,</a>
<a class="sourceLine" id="cb20-9" data-line-number="9">                 batch <span class="op">==</span><span class="st"> "Human"</span>]),</a>
<a class="sourceLine" id="cb20-10" data-line-number="10">      <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(cell_type[batch <span class="op">==</span><span class="st"> "Human"</span>]),</a>
<a class="sourceLine" id="cb20-11" data-line-number="11">      <span class="dt">method =</span> <span class="st">"spearman"</span>)) -&gt;<span class="st"> </span></a>
<a class="sourceLine" id="cb20-12" data-line-number="12"><span class="st">  </span>corr_wrt_devel</a>
<a class="sourceLine" id="cb20-13" data-line-number="13"></a>
<a class="sourceLine" id="cb20-14" data-line-number="14"><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/qplot.html">qplot</a></span>(corr_wrt_devel[,<span class="dv">1</span>],</a>
<a class="sourceLine" id="cb20-15" data-line-number="15">      corr_wrt_devel[,<span class="dv">2</span>], </a>
<a class="sourceLine" id="cb20-16" data-line-number="16">      <span class="dt">xlim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>),</a>
<a class="sourceLine" id="cb20-17" data-line-number="17">      <span class="dt">ylim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>))</a>
<a class="sourceLine" id="cb20-18" data-line-number="18">}) -&gt;<span class="st"> </span>plts</a>
<a class="sourceLine" id="cb20-19" data-line-number="19"><span class="co">#&gt; Warning in cor(t(combined[gene_set, batch == "Mouse"]),</span></a>
<a class="sourceLine" id="cb20-20" data-line-number="20"><span class="co">#&gt; as.numeric(cell_type[batch == : the standard deviation is zero</span></a>
<a class="sourceLine" id="cb20-21" data-line-number="21"><span class="co">#&gt; Warning in cor(t(combined[gene_set, batch == "Human"]),</span></a>
<a class="sourceLine" id="cb20-22" data-line-number="22"><span class="co">#&gt; as.numeric(cell_type[batch == : the standard deviation is zero</span></a>
<a class="sourceLine" id="cb20-23" data-line-number="23"><span class="co">#&gt; Warning in cor(t(combined[gene_set, batch == "Mouse"]),</span></a>
<a class="sourceLine" id="cb20-24" data-line-number="24"><span class="co">#&gt; as.numeric(cell_type[batch == : the standard deviation is zero</span></a>
<a class="sourceLine" id="cb20-25" data-line-number="25"><span class="co">#&gt; Warning in cor(t(combined[gene_set, batch == "Human"]),</span></a>
<a class="sourceLine" id="cb20-26" data-line-number="26"><span class="co">#&gt; as.numeric(cell_type[batch == : the standard deviation is zero</span></a></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(<span class="dt">X =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(plts),<span class="dt">FUN =</span> <span class="cf">function</span>(gs_name){</a>
<a class="sourceLine" id="cb21-2" data-line-number="2">  p &lt;-<span class="st"> </span>plts[[gs_name]]</a>
<a class="sourceLine" id="cb21-3" data-line-number="3">  p <span class="op">+</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="st">    </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">"Mouse"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb21-5" data-line-number="5"><span class="st">    </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"Human"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="st">    </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(gs_name) <span class="op">+</span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="st">    </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="st">    </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(</a>
<a class="sourceLine" id="cb21-9" data-line-number="9">      <span class="dt">axis.title =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>()</a>
<a class="sourceLine" id="cb21-10" data-line-number="10">    ) <span class="op">+</span></a>
<a class="sourceLine" id="cb21-11" data-line-number="11"><span class="st">    </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html">coord_fixed</a></span>()</a>
<a class="sourceLine" id="cb21-12" data-line-number="12">  ggExtra<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggExtra/topics/ggMarginal">ggMarginal</a></span>(p<span class="op">+</span></a>
<a class="sourceLine" id="cb21-13" data-line-number="13"><span class="st">                        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="ot">NULL</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-14" data-line-number="14"><span class="st">                        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="ot">NULL</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-15" data-line-number="15"><span class="st">                        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb21-16" data-line-number="16"><span class="st">                        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html">coord_fixed</a></span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb21-17" data-line-number="17"><span class="st">                        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/annotate.html">annotate</a></span>(<span class="st">"text"</span>,</a>
<a class="sourceLine" id="cb21-18" data-line-number="18">                                 <span class="dt">x =</span> <span class="dv">-1</span>, </a>
<a class="sourceLine" id="cb21-19" data-line-number="19">                                 <span class="dt">y =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb21-20" data-line-number="20">                                 <span class="dt">label =</span> gs_name,</a>
<a class="sourceLine" id="cb21-21" data-line-number="21">                                 <span class="dt">hjust =</span> <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb21-22" data-line-number="22">                                 <span class="dt">color =</span> <span class="st">"red"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb21-23" data-line-number="23"><span class="st">                        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">plot.title =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="dt">vjust =</span> <span class="dv">1</span>)),</a>
<a class="sourceLine" id="cb21-24" data-line-number="24">                      <span class="dt">type =</span> <span class="st">"histogram"</span>)</a>
<a class="sourceLine" id="cb21-25" data-line-number="25">}) -&gt;<span class="st"> </span>histograms</a>
<a class="sourceLine" id="cb21-26" data-line-number="26"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb21-27" data-line-number="27"></a>
<a class="sourceLine" id="cb21-28" data-line-number="28"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb21-29" data-line-number="29"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb21-30" data-line-number="30"></a>
<a class="sourceLine" id="cb21-31" data-line-number="31"></a>
<a class="sourceLine" id="cb21-32" data-line-number="32">histograms[[<span class="dv">6</span>]] &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb21-33" data-line-number="33"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/qplot.html">qplot</a></span>(iris[, <span class="dv">1</span>], iris[, <span class="dv">2</span>], <span class="dt">shape =</span> <span class="ot">NA</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>,<span class="dt">xlim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>),</a>
<a class="sourceLine" id="cb21-34" data-line-number="34">      <span class="dt">ylim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb21-35" data-line-number="35"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">"Correlation w.r.t </span><span class="ch">\n</span><span class="st">development in mouse"</span>)<span class="op">+</span></a>
<a class="sourceLine" id="cb21-36" data-line-number="36"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"Correlation w.r.t </span><span class="ch">\n</span><span class="st">development in human"</span>)<span class="op">+</span></a>
<a class="sourceLine" id="cb21-37" data-line-number="37"><span class="st">  </span><span class="co"># theme_bw() +</span></a>
<a class="sourceLine" id="cb21-38" data-line-number="38"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(</a>
<a class="sourceLine" id="cb21-39" data-line-number="39">      <span class="co"># axis.title = element_blank(),</span></a>
<a class="sourceLine" id="cb21-40" data-line-number="40">      <span class="dt">axis.ticks =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>(),</a>
<a class="sourceLine" id="cb21-41" data-line-number="41">      <span class="dt">axis.text =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>(),</a>
<a class="sourceLine" id="cb21-42" data-line-number="42">      <span class="dt">plot.margin=</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/reexports.html">unit</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>),<span class="st">"cm"</span>)</a>
<a class="sourceLine" id="cb21-43" data-line-number="43">    ) <span class="op">+</span></a>
<a class="sourceLine" id="cb21-44" data-line-number="44"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/annotate.html">annotate</a></span>(<span class="st">"text"</span>,</a>
<a class="sourceLine" id="cb21-45" data-line-number="45">           <span class="dt">x =</span> <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb21-46" data-line-number="46">           <span class="dt">y =</span> <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb21-47" data-line-number="47">           <span class="dt">label =</span> <span class="st">"Each scatter point </span><span class="ch">\n</span><span class="st">represents a gene"</span>,</a>
<a class="sourceLine" id="cb21-48" data-line-number="48">           <span class="dt">hjust =</span> <span class="fl">0.5</span>, </a>
<a class="sourceLine" id="cb21-49" data-line-number="49">           <span class="dt">vjust =</span> <span class="fl">0.5</span>)<span class="op">+</span></a>
<a class="sourceLine" id="cb21-50" data-line-number="50"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/annotate.html">annotate</a></span>(<span class="st">"text"</span>,</a>
<a class="sourceLine" id="cb21-51" data-line-number="51">           <span class="dt">x =</span> <span class="dv">-1</span>,</a>
<a class="sourceLine" id="cb21-52" data-line-number="52">           <span class="dt">y =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb21-53" data-line-number="53">           <span class="dt">label =</span> <span class="st">"Gene set name"</span>,</a>
<a class="sourceLine" id="cb21-54" data-line-number="54">           <span class="dt">hjust =</span> <span class="dv">0</span>, </a>
<a class="sourceLine" id="cb21-55" data-line-number="55">           <span class="dt">vjust =</span> <span class="fl">0.5</span>,</a>
<a class="sourceLine" id="cb21-56" data-line-number="56">           <span class="dt">color =</span> <span class="st">"red"</span>)<span class="op">+</span></a>
<a class="sourceLine" id="cb21-57" data-line-number="57"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html">coord_fixed</a></span>()</a>
<a class="sourceLine" id="cb21-58" data-line-number="58"></a>
<a class="sourceLine" id="cb21-59" data-line-number="59">histograms[[<span class="st">"labels"</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>],<span class="st">""</span>)</a>
<a class="sourceLine" id="cb21-60" data-line-number="60"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/do.call">do.call</a></span>(plot_grid,histograms) -&gt;<span class="st"> </span>devel_plt</a>
<a class="sourceLine" id="cb21-61" data-line-number="61">devel_plt</a></code></pre></div>
<p><img src="embryo-devel_files/figure-html/unnamed-chunk-23-1.png" width="576"></p>
</div>
</div>
<div id="brennecke-highly-variable-genes" class="section level1">
<h1 class="hasAnchor">
<a href="#brennecke-highly-variable-genes" class="anchor"></a>Brennecke highly variable genes</h1>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">gene_sets &lt;-<span class="st"> </span><span class="kw">get_compared_gene_sets</span>(</a>
<a class="sourceLine" id="cb22-2" data-line-number="2">  <span class="dt">batch1_top_genes =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/M3Drop/topics/Brennecke_getVariableGenes">BrenneckeGetVariableGenes</a></span>(<span class="kw">logcounts</span>(yan)),</a>
<a class="sourceLine" id="cb22-3" data-line-number="3">  <span class="dt">batch1_name =</span> <span class="st">"HVG(Human)"</span>,</a>
<a class="sourceLine" id="cb22-4" data-line-number="4">  <span class="dt">batch2_top_genes =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/M3Drop/topics/Brennecke_getVariableGenes">BrenneckeGetVariableGenes</a></span>(<span class="kw">logcounts</span>(deng)),</a>
<a class="sourceLine" id="cb22-5" data-line-number="5">  <span class="dt">batch2_name =</span> <span class="st">"HVG(Mouse)"</span>,</a>
<a class="sourceLine" id="cb22-6" data-line-number="6">  <span class="dt">desired_size =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(corgi_gene_set)</a>
<a class="sourceLine" id="cb22-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="co">#&gt; Warning in glmgam.fit(cbind(a0 = 1, a1tilde = 1/meansSp[useForFit]),</span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="co">#&gt; cv2Sp[useForFit]): Too much damping - convergence tolerance not achievable</span></a>
<a class="sourceLine" id="cb22-10" data-line-number="10"><span class="co">#&gt; Warning in xy.coords(x, y, xlabel, ylabel, log): 352 x values &lt;= 0 omitted</span></a>
<a class="sourceLine" id="cb22-11" data-line-number="11"><span class="co">#&gt; from logarithmic plot</span></a>
<a class="sourceLine" id="cb22-12" data-line-number="12"><span class="co">#&gt; Warning in xy.coords(x, y, xlabel, ylabel, log): 115 x values &lt;= 0 omitted</span></a>
<a class="sourceLine" id="cb22-13" data-line-number="13"><span class="co">#&gt; from logarithmic plot</span></a></code></pre></div>
<p><img src="embryo-devel_files/figure-html/unnamed-chunk-25-1.png" width="700"><img src="embryo-devel_files/figure-html/unnamed-chunk-25-2.png" width="700"></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"></a>
<a class="sourceLine" id="cb23-2" data-line-number="2">gene_sets[[<span class="st">"CorGI"</span>]] &lt;-<span class="st"> </span>corgi_gene_set</a>
<a class="sourceLine" id="cb23-3" data-line-number="3"></a>
<a class="sourceLine" id="cb23-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(gene_sets)</a>
<a class="sourceLine" id="cb23-5" data-line-number="5"><span class="co">#&gt; [1] "HVG(Human)"   "HVG(Mouse)"   "Union"        "Intersection"</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="co">#&gt; [5] "CorGI"</span></a></code></pre></div>
<p>Reorder the gene sets</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">gene_sets &lt;-<span class="st"> </span>gene_sets[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>)]</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(gene_sets, length)</a>
<a class="sourceLine" id="cb24-3" data-line-number="3"><span class="co">#&gt; $CorGI</span></a>
<a class="sourceLine" id="cb24-4" data-line-number="4"><span class="co">#&gt; [1] 200</span></a>
<a class="sourceLine" id="cb24-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb24-6" data-line-number="6"><span class="co">#&gt; $`HVG(Human)`</span></a>
<a class="sourceLine" id="cb24-7" data-line-number="7"><span class="co">#&gt; [1] 200</span></a>
<a class="sourceLine" id="cb24-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb24-9" data-line-number="9"><span class="co">#&gt; $`HVG(Mouse)`</span></a>
<a class="sourceLine" id="cb24-10" data-line-number="10"><span class="co">#&gt; [1] 200</span></a>
<a class="sourceLine" id="cb24-11" data-line-number="11"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb24-12" data-line-number="12"><span class="co">#&gt; $Union</span></a>
<a class="sourceLine" id="cb24-13" data-line-number="13"><span class="co">#&gt; [1] 200</span></a>
<a class="sourceLine" id="cb24-14" data-line-number="14"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb24-15" data-line-number="15"><span class="co">#&gt; $Intersection</span></a>
<a class="sourceLine" id="cb24-16" data-line-number="16"><span class="co">#&gt; [1] 200</span></a>
<a class="sourceLine" id="cb24-17" data-line-number="17">gene_sets[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/sample.html">sample</a></span>(shared_genes,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(gene_sets[[<span class="dv">1</span>]]))</a>
<a class="sourceLine" id="cb24-18" data-line-number="18"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(gene_sets)[<span class="dv">1</span>] &lt;-<span class="st"> "Random"</span></a></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">embeddings_pca &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb25-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(</a>
<a class="sourceLine" id="cb25-3" data-line-number="3">    <span class="dt">X =</span> gene_sets,</a>
<a class="sourceLine" id="cb25-4" data-line-number="4">    <span class="dt">FUN =</span> <span class="cf">function</span>(gene_set) {</a>
<a class="sourceLine" id="cb25-5" data-line-number="5">      <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/prcomp">prcomp</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(combined[gene_set,]),<span class="dt">rank. =</span> <span class="dv">2</span>)<span class="op">$</span>x</a>
<a class="sourceLine" id="cb25-6" data-line-number="6">    }</a>
<a class="sourceLine" id="cb25-7" data-line-number="7">  )</a></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">emb_name &lt;-<span class="st"> "PC"</span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2">embeddings &lt;-<span class="st"> </span>embeddings_pca</a>
<a class="sourceLine" id="cb26-3" data-line-number="3"></a>
<a class="sourceLine" id="cb26-4" data-line-number="4"></a>
<a class="sourceLine" id="cb26-5" data-line-number="5"><span class="kw">get_scatterplots</span>(embeddings,</a>
<a class="sourceLine" id="cb26-6" data-line-number="6">                 <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">as.factor</a></span>(batch),</a>
<a class="sourceLine" id="cb26-7" data-line-number="7">                 cell_type) -&gt;<span class="st"> </span>dr_plots</a>
<a class="sourceLine" id="cb26-8" data-line-number="8"></a>
<a class="sourceLine" id="cb26-9" data-line-number="9"><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb26-10" data-line-number="10">  <span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb26-11" data-line-number="11">    <span class="kw"><a href="../reference/get_axes_legend.html">get_axes_legend</a></span>(emb_name),</a>
<a class="sourceLine" id="cb26-12" data-line-number="12">    <span class="kw">get_shape_legend</span>(batch, my_shape_palette), <span class="dt">nrow =</span> <span class="dv">2</span></a>
<a class="sourceLine" id="cb26-13" data-line-number="13">  ),</a>
<a class="sourceLine" id="cb26-14" data-line-number="14">  <span class="kw">get_color_legend</span>(cell_type, my_color_palette,<span class="dt">ncol =</span> <span class="dv">1</span>),</a>
<a class="sourceLine" id="cb26-15" data-line-number="15">  <span class="dt">nrow =</span> <span class="dv">1</span>) -&gt;<span class="st"> </span>legends</a>
<a class="sourceLine" id="cb26-16" data-line-number="16"></a>
<a class="sourceLine" id="cb26-17" data-line-number="17">dr_plots[[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(dr_plots)<span class="op">+</span><span class="dv">1</span>]] &lt;-<span class="st"> </span>legends</a>
<a class="sourceLine" id="cb26-18" data-line-number="18"></a>
<a class="sourceLine" id="cb26-19" data-line-number="19">dr_plots[[<span class="st">"nrow"</span>]] &lt;-<span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb26-20" data-line-number="20"></a>
<a class="sourceLine" id="cb26-21" data-line-number="21">dr_plots[[<span class="st">"labels"</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"A"</span>,<span class="st">"B"</span>,<span class="st">"C"</span>,<span class="st">"D"</span>,<span class="st">"E"</span>,<span class="st">""</span>)</a>
<a class="sourceLine" id="cb26-22" data-line-number="22"></a>
<a class="sourceLine" id="cb26-23" data-line-number="23"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/do.call">do.call</a></span>(plot_grid,dr_plots) -&gt;<span class="st"> </span>top_panel</a>
<a class="sourceLine" id="cb26-24" data-line-number="24">top_panel</a></code></pre></div>
<p><img src="embryo-devel_files/figure-html/unnamed-chunk-28-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#preprocessing-the-data">Preprocessing the data</a></li>
      <li><a href="#run-corgi-feature-selection">Run CorGI feature selection</a></li>
      <li>
<a href="#visualization">Visualization</a><ul class="nav nav-pills nav-stacked">
<li><a href="#principal-component-analysis-pca">Principal component analysis (PCA)</a></li>
      </ul>
</li>
      <li><a href="#brennecke-highly-variable-genes">Brennecke highly variable genes</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Yutong Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
